### ISSUE TITLE
issue with silx view and relative path ?

### ISSUE THREAD

**Issue by payno** [silx-kit/silx#3277]
a scientist had a strange behavior with the current silx view (has an environment with the dev version. Last commit is 2ecfdeb90b76d3008b3040a23b1e01a6b07b7c93)

Anyway this is the scenario to reproduce:

```bash
cd /data/visitor/ls2914/id19/set40_tomo_black_drum_LPJ21_6p5mu_unload/set40_tomo_black_drum_LPJ21_6p5mu_unload_0001
silx view set40_tomo_black_drum_LPJ21_6p5mu_unload_0001.h5
cd ../../set43_tomo_black_drum_LPJ05_6p5mu_0N/set43_tomo_black_drum_LPJ05_6p5mu_0N_0003/
silx view ../../set40_tomo_black_drum_LPJ21_6p5mu_unload/set40_tomo_black_drum_LPJ21_6p5mu_unload_0001/set40_tomo_black_drum_LPJ21_6p5mu_unload_0001.h5
```

But the display of frames differs: 

first silx view
![image](https://user-images.githubusercontent.com/12907796/99779534-79a38800-2b15-11eb-8ee8-5e3bcb90aec3.png)
![image](https://user-images.githubusercontent.com/12907796/99779685-b96a6f80-2b15-11eb-89d4-5ae6393d4354.png)
![image](https://user-images.githubusercontent.com/12907796/99779732-c5eec800-2b15-11eb-9353-a05b6ecb9e99.png)

second silx view
![image](https://user-images.githubusercontent.com/12907796/99779562-832cf000-2b15-11eb-956f-5ea2677cfaea.png)
![image](https://user-images.githubusercontent.com/12907796/99779613-99d34700-2b15-11eb-93b1-0b94d2de5e5a.png)
![image](https://user-images.githubusercontent.com/12907796/99779630-a2c41880-2b15-11eb-9bb0-b08ee21df553.png)

file name are identicals on the silx view.
"/data/visitor/ls2914/id19/set40_tomo_black_drum_LPJ21_6p5mu_unload/set40_tomo_black_drum_LPJ21_6p5mu_unload_0001/set40_tomo_black_drum_LPJ21_6p5mu_unload_0001.h5"
"/data/visitor/ls2914/id19/set40_tomo_black_drum_LPJ21_6p5mu_unload/set40_tomo_black_drum_LPJ21_6p5mu_unload_0001/set40_tomo_black_drum_LPJ21_6p5mu_unload_0001.h5"

It seems this not append when using the silx version from rnice (not tested personally). Same behavior with 0.13

**Comment by payno** [comment]
I find For information the 'frelon2' are virtual dataset (with relative path).

**Comment by t20100** [comment]
Reproduced on debian9 with silx 0.13.2, h5py 2.70. libhdf5 1.10.0 (arg !)..

Apparently what is displayed in the 2nd case with relative path is the same as what's in `set43_tomo_black_drum_LPJ05_6p5mu_0N_0003.h5` ;(
That points to an issue with current working directory/relative path in links/VDS...

**Comment by payno** [comment]
maybe a simple change of cwd when calling get_data is sufficient ?

**Comment by t20100** [comment]
Maybe, but you will end-up with changing cwd all the time if multiple files are opened... and how does this behaves with multiple external links/VDS?

**Comment by payno** [comment]
no I think all links should be done if relative according to the file. Then this is HDF5 to manage the reading

**Comment by payno** [comment]
What I find a bit strange is that if I go one or two folder upper then it works. I wonder if somewhere in the code underlying (HDF5 / h5py ???) they don't check if the files exists then they take it otherwise they look at a different location ??? !!!

**Comment by t20100** [comment]
I confirm, I can reproduce the same with only calls to `h5py`.
Apparently, if the CWD contains path to files used in the VDS they are used instead of the one relative to the file!

**Comment by woutdenolf** [comment]
This test fails with `hdf5: 1.10.4, h5py: 2.10.0` and passes with `hdf5: 1.12.0, h5py: 3.0.0`.

```python
import h5py
import os
import numpy

os.makedirs("anotherfolder", exist_ok=True)


with h5py.File(os.path.join("data.h5"), "w") as f:
    f["data"] = numpy.zeros(10)


with h5py.File(os.path.join("master.h5"), "w") as f:
    layout = h5py.VirtualLayout(shape=(2,), dtype='f4')
    source = h5py.VirtualSource("data.h5", 'data', shape=(10,))
    layout[:] = source[:2]
    f.create_virtual_dataset('data', layout)


with h5py.File(os.path.join("anotherfolder/data.h5"), "w") as f:
    f["data"] = numpy.ones(10)


with h5py.File(os.path.join("master.h5"), "r") as f:
    assert f["data"][0] == 0


os.chdir("anotherfolder")
with h5py.File(os.path.join("..", "master.h5"), "r") as f:
    assert f["data"][0] == 0
os.chdir("..")
```

**Comment by vallsv** [comment]
Be brave guys...

**Comment by t20100** [comment]
Issue fixed: just use up-to-date software ;)

**Comment by woutdenolf** [comment]
The lima links have this form `scanxxxx/frelon2_0000.h5::entry_0000/instrument/frelon2/data`

So if you open a dataset file while the CWD is**another** dataset directory **and** the CWD dataset also contains scan `xxxx`, this issue occurs.

So this issue is rather obscure (opening a dataset while being in another dataset directory). In addition it is fixed by h5py 3.0 (or the underlying hdf5 version). So close imo.

**Comment by t20100** [comment]
To fix this, use libhdf5 >= 1.10.6, h5py 3.x wheels includes libhdf5 1.12.0 so it fixes the issue.
